### PR TITLE
Tzfix

### PR DESCRIPTION
**Pull request by prsauer**

### CODE REVIEW

File: packages/parser/src/index.ts [hunk 1]
```diff
@@ -28,18 +36,34 @@ export interface IParserContext {
 const WOW_VERSION_LINE_PARSER = /COMBAT_LOG_VERSION,(\d+),ADVANCED_LOG_ENABLED,\d,BUILD_VERSION,([^,]+),(.+)\s*$/;
 
 export class WoWCombatLogParser extends EventEmitter {
+  private _timezone: string;
```
Comment (kunchenguid) [inline packages/parser/src/index.ts:39]: might be able to make this a 'public readonly' sine the only place where it's modified is in the constructor?

File: packages/parser/src/pipeline/classic/segmentToCombat.ts [hunk 2]
```diff
@@ -7,11 +7,11 @@ import { CombatUnitReaction, CombatUnitType, ICombatEventSegment } from '../../t
 import { computeCanonicalHash, nullthrows } from '../../utils';
 import { isNonNull } from '../common/utils';
 
-export const segmentToCombat = () => {
+export const segmentToCombat = (timezone: string) => {
   return pipe(
     map((segment: ICombatEventSegment): IArenaMatch | IMalformedCombatData | null => {
       if (segment.events.length >= 3) {
-        const combat = new CombatData('classic');
+        const combat = new CombatData('classic', timezone);
```
Comment (kunchenguid) [inline packages/parser/src/pipeline/classic/segmentToCombat.ts:14]: can this be derived from segment.events so this function doesn't need to take the parameter?

File: packages/shared/src/utils/upload.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export async function uploadCombatAsync(
 
   const headers: Record<string, string> = {
     'content-type': 'text/plain;charset=UTF-8',
-    'x-goog-meta-wow-version': combat.wowVersion || 'retail',
+    'x-goog-meta-wow-version': combat.wowVersion,
     'x-goog-meta-ownerid': ownerId,
     'x-goog-meta-starttime-utc': combat.startTime.toString(),
```
Comment (kunchenguid) [inline packages/shared/src/utils/upload.ts:20]: need the timezone header?

Comment (prsauer) [reply packages/shared/src/utils/upload.ts:20]: It's in there, just below this

File: packages/parser/src/pipeline/classic/segmentToCombat.ts (hunk 2 above)
Comment (prsauer) [reply packages/parser/src/pipeline/classic/segmentToCombat.ts:14]: Sort of -- to factor it out we effectively need to attach it to every event which might be fine

File: packages/parser/src/index.ts [hunk 4]
```diff
@@ -28,15 +36,27 @@ export interface IParserContext {
 const WOW_VERSION_LINE_PARSER = /COMBAT_LOG_VERSION,(\d+),ADVANCED_LOG_ENABLED,\d,BUILD_VERSION,([^,]+),(.+)\s*$/;
 
 export class WoWCombatLogParser extends EventEmitter {
+  public readonly _timezone: string;
```
Comment (kunchenguid) [inline packages/parser/src/index.ts:39]: nit - since it's public might be better to remove the underscore prefix :)

**Review by kunchenguid — APPROVED** [review]

**Pull request merged**